VRoidHubLoader/Modules/VrmLoaderModule.cs

```diff
@@ -62,17 +63,37 @@ private void Update()
         {
             Core.Msg("OnUpdate: VrmLoaderModule F4 pressed");
 
-            // MenuManager is a singleton? sweet.
-            if (!MenuManager.Instance.IsOpen)
+            await ImportVRM();
```

Comment:
I know it doesn't matter in this case as the Update loop isn't doing anything else, but do we really want to await in Update? Isn't this more of a fire and forget that we don't want to await here.

Reply:
To my knowledge we kinda have to await here to prevent the file selection window from blocking the thread. I already tried doing this inside of a coroutine and it froze the app. If there's a way to do this without awaiting then I agree.

Reply:
Interesting, even if you call ImportVRM without the await statement and keeping ImportVRM as an async function?

I would think that would just fire and forget like JS handles async functions called without await.

Reply:
I just tried it and the thread does not get blocked if `ImportVRM()` is not awaited, however the current production code does use an `await` before I moved it into a helper function, so we can probably keep it as is if it isn't doing any harm.

Reply:
That was code I wrote, but the await was added before merge and I was at work so I didn't see that change until it was already merged, and it wasn't a big enough deal to do another PR to clean it up. I intentionally didn't await there in case more was added to the update loop, as you don't want to block that potential future code, but I probably should have added a comment making that clear.

Reply:
Ah, thanks for clarifying. Since it seems to work as a fire-and-forget, I can modify it to run that way in a future commit before this is merged.

Reply:
Changed in [9bcf308](https://github.com/YusufOzmen01/desktopmate-custom-avatar-loader/pull/82/commits/9bcf3089a008107f7b725e3acdc92a9ebd88118c)